### PR TITLE
chore: add workflow for publishing npm package

### LINKED ISSUE #3: Create a workflow for mento-sdk NPM package generation
**Description**

Write a script in `mento-sdk/.github/workflows` similar to the one in [mento-core-ts](https://github.com/mento-protocol/mento-core-ts/blob/main/.github/workflows/publish_npm.yml) to allow npm package deployment from GitHub actions

Refer to the [release a versioning doc](https://www.notion.so/Release-process-and-versioning-78bd7021a39a4fdf8e7495174a19d676) for more context 

**Acceptance Criteria**

- [ ] Workflow script is written and works as expected
- [ ] One can trigger the workflow manually from GitHub actions
- [ ] Version 0.1.0 is successfully deployed on mentolabs NPM org (everyone should be invited/part of the org that @ninocomputer created, check if you're a part of it or request a new invitation)
- [ ] Draft a [github release](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository) with the same tag to publish recent code changes

### PR DESCRIPTION
**Pull request by ninocomputer**
### Description

Added a workflow for publishing mento-sdk npm package and updated the existing package to include new license and updated readme.

### Other changes

n/a

### Tested

[Package was successfully published ](https://www.npmjs.com/package/@mento-protocol/mento-sdk?activeTab=readme)

### Related issues

- Fixes #3 

### Backwards compatibility

n/a

### Documentation

n/a

### CODE REVIEW

File: .github/workflows/publish_npm.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+jobs:
+  release:
+    name: publish
+    runs-on: ubuntu-latest
+    steps:
+      - name: 'Check out the repo'
+        uses: 'actions/checkout@v3'
+        with:
+          submodules: 'recursive'
```
Comment (jmrossy) [inline .github/workflows/publish_npm.yml:13]: Not necessary unless we're using git submodules in the repo

File: .github/workflows/publish_npm.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: 'Install the Node.js dependencies'
+        run: 'yarn install --immutable'
+
+      - name: 'Build the sdk'
+        run: 'yarn build'
+
+      - name: 'Build types'
+        run: 'yarn build'
```
Comment (jmrossy) [inline .github/workflows/publish_npm.yml:29]: Duplicate commands here

Comment (ninocomputer) [reply .github/workflows/publish_npm.yml]: updated 👍🏼

File: .github/workflows/publish_npm.yml (hunk 1 above)
Comment (jmrossy) [reply .github/workflows/publish_npm.yml]: Sorry I wasn't clear. You do need the `'actions/checkout@v3'` step, you just don't need the with submodules: 'recursive' line.

**Comment by jmrossy** [comment]
Sorry I wasn't clear. You do need the 'actions/checkout@v3' step, you just don't need the with submodules: 'recursive' line.

**Comment by ninocomputer** [comment]
> Sorry I wasn't clear. You do need the 'actions/checkout@v3' step, you just don't need the with submodules: 'recursive' line.

Sorry from my side as well, I have a cold and it shows 🤦🏻‍♀️ fixing it now 👍🏼

**Review by jmrossy — APPROVED** [review]

File: .github/workflows/publish_npm.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: 'Check out the repo'
+        uses: 'actions/checkout@v3'
+        
+      - name: 'Install Node.js'
+        uses: 'actions/setup-node@v3'
+        with:
+          cache: 'yarn'
+          node-version: '16'
```
Comment (nvtaveras) [inline .github/workflows/publish_npm.yml:17]: should we use 18 here as it's the current LTS?

Comment (ninocomputer) [reply .github/workflows/publish_npm.yml:17]: sure

**Pull request merged**